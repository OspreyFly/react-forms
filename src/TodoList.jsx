import React, { useState } from 'react';
import NewTodoForm from './NewTodoForm';
import Todo from './Todo';

function TodoList() {
    const [todos, setTodos] = useState([]);

    const addTodo = (task) => {
        const newTodos = [...todos, { task, id: Date.now() }];
        setTodos(newTodos);
    };

    const removeTodo = (id) => {
        const newTodos = todos.filter(todo => todo.id !== id);
        setTodos(newTodos);
    };

    return (
        <div>
            <NewTodoForm addTodo={addTodo} />
            {todos.map(todo => (
                <Todo key={todo.id} todo={todo} removeTodo={removeTodo} />
            ))}
        </div>
    );
}

export default TodoList;
