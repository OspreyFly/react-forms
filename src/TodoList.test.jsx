// TodoList.test.js
import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import TodoList from './TodoList';

test('renders without crashing', () => {
    render(<TodoList />);
});

test("matches snapshot", () => {
    const {asFragment} = render(<TodoList />);
    expect(asFragment()).toMatchSnapshot();
})

test('adds a new todo', () => {
 const { queryByText, getByText } = render(<TodoList />);
 const input = getByText("Add a new task");
 const btn = getByText("Add");
 expect(queryByText('Walk the dog')).not.toBeInTheDocument();
 fireEvent.change(input, { target: { value: 'Walk the dog' }});
 fireEvent.click(btn);
 expect(queryByText('Walk the dog')).toBeInTheDocument();
});
