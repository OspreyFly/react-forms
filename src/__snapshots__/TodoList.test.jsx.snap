// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <input
        placeholder="Add a new task"
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Add
      </button>
    </form>
  </div>
</DocumentFragment>
`;
